### PR TITLE
Conformance test

### LINKED ISSUE #55: Conformance test
We need a conformance test to check if other implementations works. We need it to test https://github.com/knative-sandbox/eventing-kafka-broker/pull/657

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #55 

Depends on #57 

# Changes

- :gift: Add a conformance test that sends and receives 3 messages, one that is accepted with ack, one that is accepted with an error ack, and one without payload

### CODE REVIEW

File: test/conformance/feature/conformance_feature.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package feature
```
Comment (matzew) [inline test/conformance/feature/conformance_feature.go:1]: should we do some headers? 

and do we need instructions, like:

```
// +build e2e

```

here too?

Comment (slinkydeveloper) [reply test/conformance/feature/conformance_feature.go:17]: We don't need e2e build tags I think (at least, i don't see them in eventing).

I added the copyright headers

File: test/conformance/resources/conformance_client/job.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .name }}
+  namespace: {{ .namespace }}
+  labels:
+    app: eventshub-{{ .name }}
```
Comment (matzew) [inline test/conformance/resources/conformance_client/job.yaml:21]: is that the thing from the testing framework?

Comment (slinkydeveloper) [reply test/conformance/resources/conformance_client/job.yaml:21]: yeah, the testing framework mixes golang and yaml

**Review by matzew — APPROVED** [review]
nice,
thanks for adding the client / server tests

/lgtm
/approve

**Pull request merged**